### PR TITLE
Silence delete compile warning in memory.h/cpp on Windows

### PR DESCRIPTION
**Pull request by BastiaanOlij**
Fix memory warnings during compile on Windows MSVR

### CODE REVIEW

File: src/core/memory.cpp [hunk 1]
```diff
@@ -51,3 +51,25 @@ void Memory::free_static(void *p_ptr) {
 void *operator new(size_t p_size, const char *p_description) {
 	return godot::Memory::alloc_static(p_size);
 }
+
+using namespace godot;
+
+#ifdef _MSC_VER
+void operator delete(void *p_mem, const char *p_description) {
+
```
Comment (vnen) [inline src/core/memory.cpp]: Just a style nitpick: it should not have an empty line between the function signature and the body. Same for the others down below.

File: src/core/memory.cpp [hunk 2]
```diff
@@ -51,3 +51,25 @@ void Memory::free_static(void *p_ptr) {
...
+	ERR_PRINT("Call to placement delete should not happen.");
+	CRASH_NOW();
+}
+
+void operator delete(void *p_mem, void *p_pointer, size_t check, const char *p_description) {
+
+	ERR_PRINT("Call to placement delete should not happen.");
+	CRASH_NOW();
+}
+#endif
```
Comment (vnen) [inline src/core/memory.cpp]: Another style nitpick: the file should end with a newline.

File: src/core/memory.cpp (hunk 1 above)
Comment (BastiaanOlij) [reply src/core/memory.cpp]: that's what I get for copy pasting from Godot source:) I'll fix it up :)

File: src/core/memory.cpp (hunk 2 above)
Comment (BastiaanOlij) [reply src/core/memory.cpp]: always

**Pull request merged**